addon/state-for.js

```diff
@@ -48,11 +56,15 @@ export default function stateFor(stateName, propertyName) {
     let state = weakMaps[stateName];
 
     if (!state.has(propertyValue)) {
-      let newState = createStateFor(this, stateName, this.container);
-      state.set(propertyValue, newState);
+      state.set(propertyValue, {
+        [guidString(secondaryKeys)]: createStateFor(this, stateName)
```

Comment:
seems like we could `let guid = guidString(secondaryKeys)` and use `guid` throughout the function.